### DISCUSSION TITLE
Encrypted Output

### DISCUSSION THREAD

**Discussion by http403** [florisboard/florisboard#2282] [Polls]
UK is proposing two new laws, Online Safety Bill and Investigative Powers Act 2016, which will undermine the security of chat apps currently using like Signal. The later is more problematic as it requires chat service provider to get approvals from the UK government before release, and disable said function silently upon government's request. While they are not in effect yet, once they are signed into law, I believe other countries will follow suite, which affects worldwide.

I would like to know if the Florisboard community interested in an optional encrypted output mode before making a feature request.

I'm thinking on a button on the smartbar. Upon activation, the smartbar will first ask the recipient, then become a text field for input, with a decrypt button. When hit with Enter, those text will be converted to encrypted message and autotype on the chatapp which might need some user interactions (like tap to focus the chat input). Decryption will done via copy, and upon decryption button actvation Florisboard will read the clipboard and output the decrypted text to the text field in non edit mode.

I know the usability is quite low as other had attempted and failed. I know there are other open source chat service with self hosting and E2EE capability. However, I would like to have a backup in case there are no safe options left which is out of what the legislation covers.

P.S. I don't code in Kotlin, nor have any knowledge in Android app developement. What I said is just imagination which I don't know if its possible. This is more like a pre- feature request to gague the support for such idea.

**Comment by inson1** [comment]
yes, idk what implementation idea is better

**Comment by rlatapy-luna** [comment]
We implemented an encrypted output on top of florisboard in our mobile application (oneSafe6). On florisboard side, we just add some stuff to be able to inject UI from our app and to be able to intercept inputs to redirect it to the keyboard textfield. Don't hesitate to take a look if you're interested.
fork: https://github.com/florisboard/florisboard/compare/master...LunabeeStudio:library
app integration: https://github.com/LunabeeStudio/oneSafe6_SDK_Android/tree/master/app/ime/src/main/kotlin/studio/lunabee/onesafe/ime
Regarding the crypto, we use the signal protocol with an initialization based on deeplink/qrcode exchange, no server.
